src/Frontend/util/use-debounced-input.ts

```diff
@@ -0,0 +1,21 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+import { useEffect, useState } from 'react';
+
+export function useDebouncedInput<T>(input: T, delay?: number): T {
```

Comment:
without a positive default delay this function really does not make much sense. this will then allow us to avoid defining this constant everywhere where we want to use the hook.
```suggestion
export function useDebouncedInput<T>(input: T, delay = 500): T {
```

Reply:
also, this might be a better default: https://humanbenchmark.com/tests/reactiontime

Reply:
Done. Had do rewrite this a little bit to avoid a no-magic-numbers linter error.